projects/step-frontend/src/lib/os-plugins/modules/script-editor/components/script-editor/script-editor.component.ts

```diff
@@ -63,17 +76,43 @@ export class ScriptEditorComponent implements AfterViewInit, OnDestroy {
       this._keywordEditorApi.getFunctionScript(this._functionId),
     ]).subscribe(([keyword, keywordScript]) => {
       this.keyword = keyword;
+      this.initialScript = keywordScript;
       this.editor!.getSession().setMode(this.determineKeywordMode());
       this.editor!.setValue(keywordScript);
+      this.focusOnText();
+      this.trackChanges();
+    });
+  }
+
+  private trackChanges(): void {
+    this.editor!.on('change', (value) => {
```

Comment:
If we subscribe to some event, we need to be sure unsubscribe the listener, when component is destroyed.
Not unsubscribed listeners, may prevent garbage collector to remove objects from memory and cause memory leaks

Reply:
Can I do that here considering this.editor!.on function returns void?

Reply:
You can do it like this:

this.editor!.off('change',...
https://ace.c9.io/api/interfaces/ace.Ace.EventEmitter.html#off.off-1